app/lib/nomis_client/base.rb

```diff
@@ -11,6 +11,7 @@ def get(path, params = {})
       end
 
       def post(path, params = {})
+        params[:headers] = update_json_headers(params[:headers]) if params[:headers]
```

Comment:
Hi @ngw, should this add/update the headers regardless of their presence when this method is called? I think I'd  expect to be able to call this with no `params[:headers]` specified and still see the headers added to the call.

Reply:
why would you add an instance variable for this? what's the point?

Reply:
Sorry, I don't think I'm suggesting that. My understanding is that the code currently will *only* call update_json_headers` when `params[:headers]` is set. The behaviour I think I'd expect is to always call that method... so if `post` is called with a `params` hash that doesn't contain `:headers` the json headers will still be added.

Reply:
Oh gotcha. Sorry misunderstood. See if you're happy now.

Reply:
ah wait, ok I see the point now

Reply:
we should be there now